src/virtual-repeat.js

```diff
@@ -271,35 +271,39 @@ export class VirtualRepeat extends AbstractRepeater {
   _getMore(): void {
     if (this.isLastIndex || this._first === 0) {
       if (!this._calledGetMore) {
-        let getMoreFuncName = this.view(0).firstChild.getAttribute('virtual-repeat-next');
-        if (!getMoreFuncName) {
-          return;
-        }
-        let getMoreFuncParts = getMoreFuncName.split('.');
-        let getMoreFunc = this.scope.overrideContext.bindingContext || {};
-        for (let propertyIndex = 0; propertyIndex < getMoreFuncParts.length; propertyIndex++) {
-          getMoreFunc = getMoreFunc[getMoreFuncParts[propertyIndex]];
-          if (!getMoreFunc) {
-            throw new Error("Can't locate 'virtual-repeat-next' function");
-          }
-        }
         let executeGetMore = () => {
           this._calledGetMore = true;
-          if (getMoreFunc instanceof Promise) {
-            return getMoreFunc.then(() => {
-              this._calledGetMore = false; //Reset for the next time
-            });
-          } else if (typeof getMoreFunc === 'function') {
-            let result = getMoreFunc.bind(this.scope.overrideContext.bindingContext)(this._first, this._bottomBufferHeight === 0, this._isAtTop);
-            if (!(result instanceof Promise)) {
-              this._calledGetMore = false; //Reset for the next time
-            } else {
-              return result.then(() => {
+          let func = this.view(0).firstChild.au['virtual-repeat-next'].instruction.attributes['virtual-repeat-next'];
+          let topIndex = this._first;
+          let isAtBottom = this._bottomBufferHeight === 0;
+          let isAtTop = this._isAtTop;
+
+          this.scope.overrideContext.topIndex = topIndex;
+          this.scope.overrideContext.isAtBottom = isAtBottom;
+          this.scope.overrideContext.isAtTop = isAtTop;
+
+          if (typeof func === 'string') {
+            let getMoreFuncName = this.view(0).firstChild.getAttribute('virtual-repeat-next');
+            let funcCall = this.scope.overrideContext.bindingContext[getMoreFuncName];
+
+            if (funcCall instanceof Promise) {
+              return funcCall.then(() => {
                 this._calledGetMore = false; //Reset for the next time
               });
+            } else if (typeof funcCall === 'function') {
+              let result = funcCall.bind(this.scope.overrideContext.bindingContext)(topIndex, isAtBottom, isAtTop);
```

Comment:
Can't we call it like
```
let result = funcCall.bind(this.scope.overrideContext.bindingContext)(topIndex, isAtBottom, isAtTop);

let result = funcCall({
    topIndex,
    isAtBottom,
    isAtTop
});
```
The `.call` binding should preserve the context. Otherwise use `funcCall.call(scope, params)`.